PerimeterXModule/Internals/Helpers/PxCommonUtils.cs

```diff
@@ -68,5 +71,43 @@ public static void AddHeaderToRequest(HttpContext context, string key, string va
 				ro.SetValue(headers, true, null);
 			}
 		}
+
+		public static bool IsEmailAddress(string str)
+		{
+			return Regex.IsMatch(str, PxConstants.EMAIL_ADDRESS_REGEX, RegexOptions.IgnoreCase);
+        }
+
+		public static string Sha256(string str)
+		{
+			using (SHA256 sha256 = SHA256.Create())
+			{
+				byte[] inputBytes = Encoding.UTF8.GetBytes(str);
+				byte[] hash = sha256.ComputeHash(inputBytes);
+				return BitConverter.ToString(hash).Replace("-", "").ToLowerInvariant();
+			}
+		}
+
+		public static string ExtractValueFromNestedJson(string pathToValue, dynamic jsonObject)
+		{
+            const char CREDENTIAL_FIELD_NESTING_DELIMITER = '.';
```

Comment:
I'd make this a default parameter so it can be changed if needed.

Also let's change the variable names here to be more generic and less CI-specific (since it's in PxCommonUtils)